entrypoint.py

```diff
@@ -0,0 +1,49 @@
+#! /usr/bin/env python
+
+"""
+    This is a Docker entrypoint that configures the container to run
+    as the same uid of the user on the host container, rather than
+    the Docker default of root. Aside from following security best
+    practices, this makes it so that any files created by the Docker
+    container are also owned by the same user on the host system.
+"""
+
+import sys
+import os
+import pwd
+import subprocess
+
+HOST_UID = os.stat('/projects').st_uid
+HOST_USER = os.environ.get('HOST_USER', 'projects_user')
+
+def does_username_exist(username):
+    try:
+        pwd.getpwnam(username)
+        return True
+    except KeyError:
+        return False
+
+def does_uid_exist(uid):
+    try:
+        pwd.getpwuid(uid)
+        return True
+    except KeyError:
+        return False
+
+if __name__ == '__main__':
+    if HOST_UID != os.geteuid():
+        if not does_uid_exist(HOST_UID):
+            username = HOST_USER
+            while does_username_exist(username):
+                username += '0'
+            home_dir = '/home/%s' % username
+            #print "Creating username %s with UID %d" % (username, HOST_UID)
```

Comment:
nit: can we delete this line? i try not to check in comments like this to keep things tidier, but personal pref :)